projects/services-ogc/src/lib/owc/owc-json.service.ts

```diff
@@ -331,195 +460,256 @@ export class OwcJsonService {
+    const groupName = this.getLayerGroupFromFolder(resource);
+
+    /** Layers with folder property */
+    if (groupName) {
+      /** unique layergroupResources */
+      if (!uniqueGroups.includes(groupName)) {
+        uniqueGroups.push(groupName);
+        /** reverse so layer order is like in the context */
+        const includedResources = layergroupResources.filter(r => this.getLayerGroupFromFolder(r) === groupName).reverse();
+        const layerGroup$ = this.createLayerGroup(groupName, includedResources, context, targetProjection);
+        return layerGroup$;
+      } else {
+        return of<null>(null);
       }
+    } else {
+      /** Single Layers */
+      const layer$ = this.createLayerFromDefaultOffering(resource, context, targetProjection);
+      return layer$;
     }
-
-    return forkJoin(layers$).pipe(
-      // making sure no undefined layers are returned
-      map((layers: Layer[]) => layers.filter(layer => layer))
-    );
   }
 
-  createLayerGroup(
-    groupName: string, includedResources: IOwsResource[], owc: IOwsContext, targetProjection: string): Observable<LayerGroup> {
 
+
+  /**
+   *
+   * @param groupName string | `${TFiltertypes}/string`
+   */
+  createLayerGroup(groupName: string, includedResources: IOwsResource[], owc: IOwsContext, targetProjection: string): Observable<LayerGroup | Layer> {
     const layers$: Observable<Layer>[] = [];
+    let filterType = null;
     for (const resource of includedResources) {
+      filterType = this.getFilterType(resource);
       layers$.push(this.createLayerFromDefaultOffering(resource, owc, targetProjection));
     }
 
-    const layerGroup$ = forkJoin(layers$).pipe(
+    const layerGroup$ = forkJoin(layers$)
       // making sure no undefined layers are returned
-      map((layers: Layer[]) => layers.filter(layer => layer)),
+      .pipe(map((layers: Layer[]) => layers.filter(layer => layer)))
       // putting layers in a LayerGroup
-      map((layers: Layer[]) => {
-        const parts = groupName.split('/');
-        const groupNameLast = parts[parts.length - 1];
-        const layerGroup = new LayerGroup({
-          id: groupName,
-          name: groupNameLast,
-          layers,
-          visible: !!Math.max(...layers.map(l => +l.visible)),
-          filtertype: layers[0].filtertype  // @TODO: can some layers have a different filter-type?
-        });
-
-        return layerGroup;
-      })
-    );
+      .pipe(map((layers: Layer[]) => {
+        if (layers.length) {
+          /** if filterType is Baselayers -> create a merged Layer */
+          if (filterType === Filtertypes.Baselayers) {
+            const mergedDescription = layers.map(i => i.description).filter(d => d); // filter empty elements
+            const legendImages = layers.map(i => i.legendImg).filter(d => d);
+            const layerOptions: ILayerOptions = {
+              type: 'custom',
+              id: `${groupName}_${layers.map(i => i.id).join(' ')}`.replace(/\s/g, '_'),
```

Comment:
For mergdLayers and Groups we try to create the ID from the other layers.
This could lead to long IDs if there are a lot of layers with long IDs.

Reply:
maybe some generic ID generation approach for all layers, with fixed length? E.g. SHA string of the layer name(s)?

Reply:
I think this generation of ID's is only used, when creating group layers from OWC Resources.

The problem here is, that I don't want to use the `groupName` as ID because we could have groups with the same name.

But generally the layers should use their own configured ID's, otherwise functions like `getLayerByKey()` or `getLayerById()` are not usable anymore.